### ISSUE TITLE
Pass bash completion does not work with the otp command

### ISSUE THREAD

**Issue by q1x0te** [tadfisher/pass-otp#28]
Hi,

Pass bash completion does not work when using the otp command in Ubuntu (with other non pass-otp commands it does work).

I am using pass v1.7 and pass-otp v1.0.0.

Thanks

**Comment by tadfisher** [comment]
I wouldn't expect it to work unless the completion function was extension-aware, and it's not.

I'd report this is a general issue to the pass mailing list, as there's nothing this project can do to enable completion without support from upstream.

**Comment by bexelbie** [comment]
Could we modify the installation procedure to edit the bash completion information provided by upstream? It seems a `sed` of that file could get us the completion, unless we can get upstream to support generalized plugin completion somehow.

This would require us to do some level of integration testing against new versions of pass though in case this file changed and broke the sed.

**Comment by tadfisher** [comment]
No, I'm not going to modify files packaged by another application.

I'll work on generalizing completion in `pass` itself.

**Comment by innir** [comment]
Just linking the corresponding issues of other extensions: https://github.com/roddhjav/pass-update/issues/9 and https://github.com/palortoff/pass-extension-tail/issues/4

**Comment by GFdevelop** [comment]
I'm vote for this iussue...**pass-extension-tail** have found a solution for this, see here https://github.com/palortoff/pass-extension-tail/blob/master/completion/pass-tail.bash.completion

**Comment by micah** [comment]
Hey @tadfisher !

I'm wondering if you made any progress in generalizing completion in `pass` as you mentioned last year in your comment? 

The thing that drives me nuts with pass otp is the lack of tab completion!

**Comment by bexelbie** [comment]
Looks like the bash completion added in 1.7.3 needs to be updated as the extension method isn't working.  I see a patch proposed from Feb 2 on the mailing list but no action, so I bumpted it.

**Comment by NilsIrl** [comment]
This now seems to be possible:

* https://github.com/palortoff/pass-extension-tail/issues/4#issuecomment-417101895
* https://github.com/roddhjav/pass-update/commit/e81139302fd7597393e39e56742b00440482f4e4
* https://git.zx2c4.com/password-store/commit/?id=6a6222eef23d25bec90015e49bd69c42062434ee

**Comment by NilsIrl** [comment]
@tadfisher You haven't been active in a long time (in regards to pass-otp), what should we do going forward?

* Give merge access to someone else? So that at least PRs can get through.?
* Should we fork?

**Comment by bexelbie** [comment]
Perhaps we can add more committers @tadfisher ?

**Comment by heavyimage** [comment]
This is still broken I think?  I'll source the otp completion manually for now but would be cool if this just worked!

**Comment by ghost** [comment]
Is this task dead? 

The bash completion is still missing in pass-otp...

**Comment by xundeenergie** [comment]
I've just created a pull-request for pass itself
https://github.com/zx2c4/password-store/pull/5
